commands/webhook.go

```diff
@@ -0,0 +1,472 @@
+	}
+
+	return nil
+}
+
+func createOutgoingWebhookCmdF(c client.Client, command *cobra.Command, args []string) error {
+	printer.SetSingle(true)
+
+	teamArg, errTeam := command.Flags().GetString("team")
+	if errTeam != nil || teamArg == "" {
+		return errors.New("Team is required")
+	}
+	team := getTeamFromTeamArg(c, teamArg)
+	if team == nil {
+		return errors.New("Unable to find team: " + teamArg)
+	}
+
+	userArg, errUser := command.Flags().GetString("user")
+	if errUser != nil || userArg == "" {
+		return errors.New("User is required")
+	}
+	user := getUserFromUserArg(c, userArg)
+	if user == nil {
+		return errors.New("Unable to find user: " + userArg)
+	}
+
+	displayName, errName := command.Flags().GetString("display-name")
+	if errName != nil || displayName == "" {
+		return errors.New("Display name is required")
+	}
+
+	triggerWords, errWords := command.Flags().GetStringArray("trigger-word")
+	if errWords != nil || len(triggerWords) == 0 {
+		return errors.New("Trigger word or words required")
+	}
+
+	callbackURLs, errURL := command.Flags().GetStringArray("url")
+	if errURL != nil || len(callbackURLs) == 0 {
+		return errors.New("Callback URL or URLs required")
+	}
+
+	triggerWhenString, _ := command.Flags().GetString("trigger-when")
+	var triggerWhen int
+	if triggerWhenString == "exact" {
+		triggerWhen = 0
+	} else if triggerWhenString == "start" {
+		triggerWhen = 1
+	} else {
+		return errors.New("Invalid trigger when parameter")
+	}
+	description, _ := command.Flags().GetString("description")
+	contentType, _ := command.Flags().GetString("content-type")
+	iconURL, _ := command.Flags().GetString("icon")
+
+	outgoingWebhook := &model.OutgoingWebhook{
+		CreatorId:    user.Id,
+		Username:     user.Username,
+		TeamId:       team.Id,
+		TriggerWords: triggerWords,
+		TriggerWhen:  triggerWhen,
+		CallbackURLs: callbackURLs,
+		DisplayName:  displayName,
+		Description:  description,
+		ContentType:  contentType,
+		IconURL:      iconURL,
+	}
+
+	channelArg, _ := command.Flags().GetString("channel")
+	if channelArg != "" {
+		channel := getChannelFromChannelArg(c, channelArg)
+		if channel != nil {
+			outgoingWebhook.ChannelId = channel.Id
+		}
+	}
+
+	createdOutgoing, respWebhookOutgoing := c.CreateOutgoingWebhook(outgoingWebhook)
+	if respWebhookOutgoing.Error != nil {
+		return respWebhookOutgoing.Error
+	}
+
+	printer.Print("Id: " + createdOutgoing.Id)
+	printer.Print("Display Name: " + createdOutgoing.DisplayName)
+
+	return nil
+}
+
+func modifyOutgoingWebhookCmdF(c client.Client, command *cobra.Command, args []string) error {
+	printer.SetSingle(true)
+
+	if len(args) < 1 {
+		return errors.New("WebhookID is not specified")
+	}
+
+	webhookArg := args[0]
+	oldHook, respWebhookOutgoing := c.GetOutgoingWebhook(webhookArg)
+	if respWebhookOutgoing.Error != nil {
+		return fmt.Errorf("unable to find webhook '%s'", webhookArg)
+	}
+
+	updatedHook := model.OutgoingWebhookFromJson(strings.NewReader(oldHook.ToJson()))
```

Comment:
What is the rationale behind transforming the outgoing webhook to JSON and back to the `struct` instead using the `struct` directly?